python/pyspark/sql/dataframe.py

```diff
@@ -1760,6 +1760,17 @@ def toPandas(self):
 
             for f, t in dtype.items():
                 pdf[f] = pdf[f].astype(t, copy=False)
+
+            if self.sql_ctx.getConf("spark.sql.execution.pandas.timeZoneAware", "false").lower() \
```

Comment:
I'd like to treat it as a bug and always respect the session local timezone.

Reply:
We still need a conf, even if it is a bug. This is just to avoid breaking any existing app. We can remove the conf in Spark 3.x.